packages/common/src/objects/deepFilter.ts

```diff
@@ -0,0 +1,39 @@
+import { isFindable } from "./internal/isFindable";
+import { setProp } from "./set-prop";
+
+/**
+ * Traverse a graph filtering out entries that do
+ * not pass the predicate
```

Comment:
Maybe add a comment with an example - see the `deepFindById` function for an example

Reply:
Also add a comment that this uses cloning behind the scenes so it only works reliably with POJO's, and not class instances